docs/concepts/callbacks.md

```diff
@@ -1,2 +1,259 @@
+
+![Example Results](https://user-images.githubusercontent.com/6599259/224284912-f3f6f547-8d75-4529-8df1-7f9904423239.png)
+
+
+## BestModelCheckpoint
+
+This callback evaluates the performance of the model at the end of each epoch, and keeps a record of the best performing model across all epochs.
+Once fitting is finished the best performing model is saved instead of the most recent model.
+The definition of best is based on two parameters:
+
+- `monitor`: The metric that is used to compare models to each other. By default this value is `val_loss`, the loss function calculated using the evaluation data, however the loss calculated on the training data can be used instead with `train_loss`; any metric that is recorded by the evaluation callback can also be used.
+- `mode`: Whether the monitored metric should be maximised (`max`) or minimised (`min`). By default the mode is set to `auto`, meaning that it will automatically choose the correct mode depending on the chosen metric: 'min' if the metric is loss and 'max' if the metric is one recorded by the evaluation callback.
+
+The console output below shows how the evaluation loss of the model is monitored between each epoch, and how the best performing model is tracked. Since the final model has a higher loss than the previously recorded best model, the best model will be saved instead of the latest one.
+
+```bash
+           INFO     Finetuning ...                                                   
+                    __main__.py:173
+[11:50:33] INFO     Model improved from inf to 2.756!                                
+       best_model_checkpoint.py:112
+[11:50:52] INFO     Model improved from 2.756 to 2.711!                              
+       best_model_checkpoint.py:112
+[11:51:10] INFO     Model did not improve                                            
+       best_model_checkpoint.py:120
+[11:51:28] INFO     Model did not improve                                            
+       best_model_checkpoint.py:120
+  Training [4/4] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54/54 0:00:00 0:00:15 • loss: 0.496 • val_loss: 2.797
+           INFO     Done ✨                                                          
+                    __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 1 minutes and 16 seconds         
+                    __main__.py:196
+           INFO     Building the artifact ...                                        
+                    __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                   
+                    __main__.py:231
+```
+
+## EarlyStopping
+
+Similarly to the best model checkpoint callback,
+the early stopping callback measures a given metric at the end of every epoch.
+Unlike the best model checkpoint callback,
+the early stopping callback does not save the best model;
+only the monitored metric is recorded between runs in order to assess the rate of improvement.
```

Comment:
```suggestion
It takes `monitor` and `mode` parameters that work the same way they do for `BestModelCheckpoint` above.
However, instead of using this information to identify the best performing model out of all epochs, it uses user-specified values at the end of each epoch to determine if it should terminate fine-tuning and save the current model.
```